### DISCUSSION TITLE
Is there any plan to support CMake's project file?

### DISCUSSION THREAD

**Discussion by Amuwa** [zed-industries/zed#16271] [Feature Requests]
When use C++ and a CMakeLists.txt importing some libraries with ```include_directories()```, it seems that ```zed``` is unable to respond to such needs and gives no hint/completion when typing classes or objects from the libraries.

Is there any plan to deal with CMakeLists.txt?

**Comment by tdimitrov** [comment]
Newbie Zed user here but I'll try to answer your question in general.
CMake is just a build system. Zed uses an lsp server/client to provide stuff like auto complete, etc. For C++ this is clangd. The latter relies on `compile_commands.json` database to work. You can generate it by passing `-DCMAKE_EXPORT_COMPILE_COMMANDS=On` when executing cmake.

TLDR: Try calling `cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=On <keys you use> ..` and see if everything works as expected.

**Comment by BrannonKing** [comment]
One additional note on this: you have to modify the .clangd file to tell it where to look for your compile_commands file. E.g:
```
CompileFlags:
  CompilationDatabase: build/clang/release
```
Also, I added the include folders into the .clangd file directly, but I think it is supposed to be smart enough to get them from the compile_commands.json file.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Reply by AviiNL** [reply]
I found this tread using a search engine, because I have an issue similar to the topic of this thread. Anything asked and answered in discord will NOT show up in a search engine.
